app/api/orders.py

```diff
@@ -116,24 +116,41 @@ def before_get_object(self, view_kwargs):
             attendee = safe_query(self, TicketHolder, 'id', view_kwargs['attendee_id'], 'attendee_id')
             view_kwargs['order_identifier'] = attendee.order.identifier
 
+        order = safe_query(self, Order, 'identifier', view_kwargs['order_identifier'], 'order_identifier')
+
+        if not has_access('is_coorganizer_or_user_itself', event_id=order.event_id, user_id=order.user_id):
+            return ForbiddenException({'source': ''}, 'Access Forbidden')
+
+
     def before_update_object(self, order, data, view_kwargs):
+
+        if not has_access('is_admin'):
+            for element in data:
+                if element != 'status':
+                    del data[element]
```

Comment:
Don't delete these, that makes them null. Just assign them their previous values again.